src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_role.py

```diff
@@ -95,6 +95,24 @@ def _build_role_io(self, flow: Union[str, PathLike], inputs_value: Dict = None):
             )
         return ChatRoleInputs(inputs), ChatRoleOutputs(outputs)
 
+    def _update_inputs_from_data_and_inputs(self, data: Dict, inputs: Dict):
+        """Update inputs from data and inputs from experiment"""
+        data_prefix = "${data."
+        inputs_prefix = "${inputs."
+        for key in self._inputs:
+            current_input = self._inputs[key]
+            value = current_input["value"]
+            if isinstance(value, str):
+                if value.startswith(data_prefix):
+                    stripped_value = value.replace(data_prefix, "").replace("}", "")
+                    data_name, col_name = stripped_value.split(".")
+                    if data_name in data:
+                        current_input["value"] = data[data_name][col_name]
```

Comment:
What do we need to do if **data[data_name][col_name]** raise a **KeyError** due to **col_name** not found?